wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -168,9 +204,40 @@ class WireJsonAdapterFactory private constructor(
       }
     }.nullSafe()
 
+    internal val UINT64_STRING_JSON_ADAPTER = object : JsonAdapter<Long>() {
+      @Throws(IOException::class)
+      override fun fromJson(reader: JsonReader): Long? {
+        return when (reader.peek()) {
+          JsonReader.Token.STRING -> {
+            UINT64_JSON_ADAPTER.fromJson(reader)
+          }
+          else -> {
+            reader.nextLong()
```

Comment:
If this works, Moshi has a bug

Reply:
This seems to work and I am not sure what the bug is. I don't think JSON can have bigger numbers than `Long.MAX_VALUE`?